In [2]:
import scipy.stats as sts
import scipy.integrate as intg
import math

1. Производится 5 выстрелов в мишень. Вероятность попадания при каждом 
выстреле равна 3/4. Найти вероятность того, что в мишени будет не менее трёх, но 
и не более четырёх пробоин. Найти наивероятнейшее число попаданий и 
соответствующую ему вероятность.

In [2]:
p_3 = sts.binom.pmf(3, 5, 3/4)
p_3

0.26367187499999983

In [3]:
p_4 = sts.binom.pmf(4, 5, 3/4)
p_4

0.3955078125

In [4]:
p_3 + p_4

0.6591796874999998

Наивероятнейшее число попаданий:

p = 3/4

q = 1 - 3/4 = 1/4

n = 5

n*p - q <= k <= n*p + p

In [5]:
a = 3/4 * 5 - 1/4
b = 3/4 * 5 + 3/4
a, b # => k = 4, вероятность уже посчитана

(3.5, 4.5)

Итак, P(3 <= x <= 4) = 0.6592 (где за x считаем кол-во попаданий); наивероятнейшее число попаданий = k = 4, p_4 = 0.3955

2. Игральная кость подбрасывается до тех пор, пока не выпадет 3 раза число очков, 
отличное от 6. Какова вероятность того, что «шестёрка» выпадет 3 раза?

A - выпала не 6

B - выпала именно 6

Тогда подходят случаи вида: AABBB (сочетания по 2 из 5) и в конце, на шестом броске, обязательно A.

In [11]:
p = math.comb(5, 2) * (5/6)**3 * (1/6)**3
p

0.026791838134430725

3. В новом микрорайоне поставлено 10000 кодовых замков на входных дверях 
домов. Вероятность выхода из строя одного замка в течение месяца равна 0,0002. 
Найти вероятность того, что за месяц откажет ровно 1 замок.

In [12]:
P = sts.poisson.pmf(1, 10000 * 0.0002)
P

0.2706705664732254

4. В банке, осуществляющем кредитование населения, 1000 клиентов. Каждому из 
клиентов выдается кредит 200 тыс. ден. ед. при условии возврата 119,31% от этой 
суммы. Вероятность невозврата кредита каждым из клиентов составляет 0,09. C 
какой вероятностью прибыль банка будет не ниже 12,8 млн. рублей?

x - кол-во тех, кто выплатили кредит

x * 200000 * 1.1931 - (1000 - x) * 200000 >= 12800000

Тогда x >= 485,... т.е x_min = 486

In [7]:
def Lap_intg(x):
    return math.exp(1)**(-(x**2/2)) * (1/math.sqrt(2*math.pi))

n = 1000
q = 0.09
p = 1- q
x1 = (486 - n*p)/(math.sqrt(n*p*q))
x2 = (1000 - n*p)/(math.sqrt(n*p*q))

f_x1 = intg.quad(Lap_intg, 0.0, x1)[0]
f_x2 = intg.quad(Lap_intg, 0.0, x2)[0]

p_x1_x2 = f_x2 - f_x1
p_x1_x2

# Ответ без учёта погрешности, в округлённом виде
# (погрешность вычислений представлена в первом элементе кортежа, который возвращает intg.quad, например:
# погрешность первого вычисления:
# intg.quad(Lap_intg, 0.0, x1)[1] = 3.2975075192543767e-13

1.0